debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/MySqlAntlrDdlParser.java

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.connector.mysql.antlr;
+
+import io.debezium.antlr.AntlrDdlParser;
+import io.debezium.antlr.AntlrDdlParserListener;
+import io.debezium.antlr.DataTypeResolver;
+import io.debezium.antlr.DataTypeResolver.DataTypeEntry;
+import io.debezium.connector.mysql.MySqlSystemVariables;
+import io.debezium.connector.mysql.antlr.listener.MySqlAntlrDdlParserListener;
+import io.debezium.ddl.parser.mysql.generated.MySqlLexer;
+import io.debezium.ddl.parser.mysql.generated.MySqlParser;
+import io.debezium.relational.Column;
+import io.debezium.relational.SystemVariables;
+import io.debezium.relational.TableEditor;
+import io.debezium.relational.TableId;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.tree.ParseTree;
+
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * A ANTLR based parser for MySQL DDL statements.
+ *
+ * @author Roman Kuchár <kucharrom@gmail.com>.
+ */
+public class MySqlAntlrDdlParser extends AntlrDdlParser<MySqlLexer, MySqlParser> {
+
+    private final ConcurrentMap<String, String> charsetNameForDatabase = new ConcurrentHashMap<>();
```

Comment:
Why is it a concurrent map? Will it ever be shared across threads (seems unlikely)?

Reply:
I reused it from existing `MySqlDdlParser`, there is also a concurrent map. But probably a good point even the existing implementation is marked as `@NotThreadSafe`